File: src/swift/palindromo.swift
//
//  palindromo.swift
//
//  Created by Matheus Torres on 14/11/20.
//

import Foundation

func isPalindrome(_ string: String) -> Bool {
    let lowercasedString: String = string.lowercased().replacingOccurrences(of: " ", with: "")
    var reversedString = ""
    
    var reversedStringArray = Array(lowercasedString)
    reversedStringArray.reverse()
    for word in reversedStringArray { reversedString += "\(word)" }
    
    return (lowercasedString == reversedString)
}

print(isPalindrome(""))
print(isPalindrome("a"))
print(isPalindrome("abba"))
print(isPalindrome("abbas"))
print(isPalindrome("tattarrattat"))
print(isPalindrome("Was it a palindrome?"))
print(isPalindrome("No lemon, no melon"))

File: src/javascript/Palindromo.js
function is_palindrome(string) {
  string = string.toLowerCase().replace(/\s/g, "");

  const reversedString = string
    .split("")
    .reverse()
    .join("")
    .toLowerCase()
    .replace(/\s/g, "");

  if (string.length <= 1) true;

  return string == reversedString;
}

function main() {
  console.log(is_palindrome(""));
  console.log(is_palindrome("a"));
  console.log(is_palindrome("abba"));
  console.log(is_palindrome("abbas"));
  console.log(is_palindrome("tattarrattat"));
  console.log(is_palindrome("Was it a palindrome?"));
  console.log(is_palindrome("No lemon, no melon"));
}

main();
